evm-adapters/src/sputnik/cheatcodes/debugger.rs

```diff
@@ -0,0 +1,403 @@
+            Opcode::PUSH32 => "PUSH32",
+            Opcode::DUP1 => "DUP1",
+            Opcode::DUP2 => "DUP2",
+            Opcode::DUP3 => "DUP3",
+            Opcode::DUP4 => "DUP4",
+            Opcode::DUP5 => "DUP5",
+            Opcode::DUP6 => "DUP6",
+            Opcode::DUP7 => "DUP7",
+            Opcode::DUP8 => "DUP8",
+            Opcode::DUP9 => "DUP9",
+            Opcode::DUP10 => "DUP10",
+            Opcode::DUP11 => "DUP11",
+            Opcode::DUP12 => "DUP12",
+            Opcode::DUP13 => "DUP13",
+            Opcode::DUP14 => "DUP14",
+            Opcode::DUP15 => "DUP15",
+            Opcode::DUP16 => "DUP16",
+            Opcode::SWAP1 => "SWAP1",
+            Opcode::SWAP2 => "SWAP2",
+            Opcode::SWAP3 => "SWAP3",
+            Opcode::SWAP4 => "SWAP4",
+            Opcode::SWAP5 => "SWAP5",
+            Opcode::SWAP6 => "SWAP6",
+            Opcode::SWAP7 => "SWAP7",
+            Opcode::SWAP8 => "SWAP8",
+            Opcode::SWAP9 => "SWAP9",
+            Opcode::SWAP10 => "SWAP10",
+            Opcode::SWAP11 => "SWAP11",
+            Opcode::SWAP12 => "SWAP12",
+            Opcode::SWAP13 => "SWAP13",
+            Opcode::SWAP14 => "SWAP14",
+            Opcode::SWAP15 => "SWAP15",
+            Opcode::SWAP16 => "SWAP16",
+            Opcode::LOG0 => "LOG0",
+            Opcode::LOG1 => "LOG1",
+            Opcode::LOG2 => "LOG2",
+            Opcode::LOG3 => "LOG3",
+            Opcode::LOG4 => "LOG4",
+            Opcode::CREATE => "CREATE",
+            Opcode::CALL => "CALL",
+            Opcode::CALLCODE => "CALLCODE",
+            Opcode::RETURN => "RETURN",
+            Opcode::DELEGATECALL => "DELEGATECALL",
+            Opcode::CREATE2 => "CREATE2",
+            Opcode::STATICCALL => "STATICCALL",
+            Opcode::REVERT => "REVERT",
+            Opcode::INVALID => "INVALID",
+            Opcode::SUICIDE => "SELFDESTRUCT",
+            _ => {
+                if let Some(cheat) = self.1 {
+                    match cheat {
+                        CheatOp::ROLL => "VM_ROLL",
+                        CheatOp::WARP => "VM_WARP",
+                        CheatOp::FEE => "VM_FEE",
+                        CheatOp::STORE => "VM_STORE",
+                        CheatOp::LOAD => "VM_LOAD",
+                        CheatOp::FFI => "VM_FFI",
+                        CheatOp::ADDR => "VM_ADDR",
+                        CheatOp::SIGN => "VM_SIGN",
+                        CheatOp::PRANK => "VM_PRANK",
+                        CheatOp::STARTPRANK => "VM_STARTPRANK",
+                        CheatOp::STOPPRANK => "VM_STOPPRANK",
+                        CheatOp::DEAL => "VM_DEAL",
+                        CheatOp::ETCH => "VM_ETCH",
+                        CheatOp::EXPECTREVERT => "VM_EXPECTREVERT",
+                        CheatOp::RECORD => "VM_RECORD",
+                        CheatOp::ACCESSES => "VM_ACCESSES",
+                        CheatOp::EXPECTEMIT => "VM_EXPECTEMIT",
+                    }
```

Comment:
I'd move this to `CheatpOp.name()` and do:

`_ => self.1.map(|cheat| cheat.name()).unwrap_or("UNDEFINED")`

Reply:
nice

Reply:
its a const func so best i can do is:
```rust

_ => {
    if let Some(cheat) = self.1 {
        cheat.name()
    } else {
        "UNDEFINED"
    }
}
```